File: 001/inverse.py
import numpy as np
import time


times = []
for _ in range(10):
    start = time.time()
    X = np.array([[0,1],[1,1],[2,1],[3,1]])
    Y = np.array([0,1,4,9])

    psuedoinverse = np.matmul(np.matmul(X.transpose(),X), np.matmul(X.transpose(),Y))
    stop = time.time()
    times.append(stop-start)

print(sum(times)/10)

File: 001/inverse.jl
points = [[0,0],[1,1],[2,4],[3,9]]
times = []
for _ in 1:10
    start = time_ns()
    X = [0 1; 1 1; 2 1; 3 1]
    Y = [0; 1; 4; 9]
    psuedoinverse = inv(transpose(X) * X) * (transpose(X) * Y)
    append!(times, time_ns()-start/1000000000)
end

print(sum(times)/10)
print("\n")